e2e/credentials-server.e2e.spec.ts

```diff
@@ -255,4 +278,34 @@ describeWithCreds('Credentials server', () => {
     // Don't bother trying to test erasing a non-existent credential, because the
     // behavior is all over the place. Fails with osxkeychain, succeeds with wincred.
   });
+
+  test.describe('should be able to detect errors', () => {
+    const bobsURL = 'https://bobs.fish/bait';
+
+    test('it should complain when no ServerURL is given', async() => {
+      const body: Record<string, string> = {};
+      const { stdout, stderr } = await rdctlCredWithStdin('store', JSON.stringify(body));
+
+      expect(stdout).toContain('no credentials server URL');
+      expect(stderr).toContain('Error: exit status 22');
+    });
+
+    test('it should complain when no username is given', async() => {
+      const body: Record<string, string> = { ServerURL: bobsURL };
+      const { stdout, stderr } = await rdctlCredWithStdin('store', JSON.stringify(body));
+
+      expect(stdout).toContain('no credentials username');
+      expect(stderr).toContain('Error: exit status 22');
+    });
+
+    test('it should not complain about other fields', async() => {
```

Comment:
```suggestion
    test('it should not complain about extra fields', async() => {
```

Should we read the result back out and see if it stores the extra field?

Reply:
sure